lib/server.js

```diff
@@ -85,15 +86,31 @@ module.exports = function createServer(config, log) {
     throw err
   }
 
-  function fetchRecords(email, ip) {
-    return P.all(
-      [
-        // get records and ignore errors by returning a new record
-        mc.getAsync(email).then(EmailRecord.parse, EmailRecord.parse),
-        mc.getAsync(ip).then(IpRecord.parse, IpRecord.parse),
-        mc.getAsync(ip + email).then(IpEmailRecord.parse, IpEmailRecord.parse)
-      ]
-    )
+  function fetchRecords(email, ip, includeReputation) {
+    var promises = [
+      // get records and ignore errors by returning a new record
+      mc.getAsync(email).then(EmailRecord.parse, EmailRecord.parse),
+      mc.getAsync(ip).then(IpRecord.parse, IpRecord.parse),
+      mc.getAsync(ip + email).then(IpEmailRecord.parse, IpEmailRecord.parse)
+    ]
+    if (includeReputation) {
+      var fetchReputation = ipClient.get(ip)
```

Comment:
Along those lines, I think it might be useful to put the reputation logical into its own class. This class could handle all of the enable checks and reporting to the reputation service. This way we could keep the `server.js` file a little cleaner.